src/pkg/artifactrash/manager.go

```diff
@@ -0,0 +1,59 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package artifactrash
+
+import (
+	"context"
+	"github.com/goharbor/harbor/src/pkg/artifactrash/dao"
+	"github.com/goharbor/harbor/src/pkg/artifactrash/model"
+)
+
+var (
+	// Mgr is a global artifact trash manager instance
+	Mgr = NewManager()
+)
+
+// Manager is the only interface of artifact module to provide the management functions for artifacts
+type Manager interface {
+	// Create ...
+	Create(ctx context.Context, artifactrsh *model.ArtifactTrash) (id int64, err error)
+	// Delete ...
+	Delete(ctx context.Context, id int64) (err error)
+	// Filter ...
+	Filter(ctx context.Context) (arts []model.ArtifactTrash, err error)
```

Comment:
should we support an option to specify whether clean up untagged artifacts?

Reply:
it's in the following PR.